### PR TITLE
EPMRPP-79645-integrationSettings-bug-fix

### PR DESCRIPTION
**Pull request by RuslanLyubimov**

### CODE REVIEW

File: app/src/components/integrations/elements/bts/btsPropertiesForIssueForm/btsPropertiesForIssueForm.jsx [hunk 1]
```diff
@@ -455,7 +455,15 @@ export class BtsPropertiesForIssueForm extends Component {
...
                   defaultOptionValueKey={this.defaultOptionValueKey}
-                />
+                  disabled={disabled}
+                >
+                  <div className={cx('show-hint-wrapper')}>
+                    <span className={cx('show-hint-text')}>
+                      {intl.formatMessage(messages.showFieldsHeader)}
+                    </span>
+                    <ShowWithTooltip />
+                  </div>
```
Comment (chivekrodis) [inline app/src/components/integrations/elements/bts/btsPropertiesForIssueForm/btsPropertiesForIssueForm.jsx:465]: Maybe better decide here should we pass children or not not directly in the component

File: app/src/components/integrations/elements/integrationSettings/integrationForm/integrationForm.jsx [hunk 2]
```diff
@@ -131,59 +131,63 @@ export class IntegrationForm extends Component {
...
             ) : (
-              <div className={cx('control-buttons-block')}>
-                <div className={cx('button-container')}>
-                  <Button
-                    onClick={handleSubmit(this.submitIntegration)}
-                    disabled={shouldFieldsBeHidden}
-                  >
-                    {formatMessage(COMMON_LOCALE_KEYS.SUBMIT)}
+              <div className={cx('integration-form-fields')}>
+                {shouldFieldsBeHidden ? null : (
```
Comment (chivekrodis) [inline app/src/components/integrations/elements/integrationSettings/integrationForm/integrationForm.jsx:143]: Can we simplify this condition to !shouldFieldsBeHidden && <>?

**Review by chivekrodis — CHANGES_REQUESTED** [review]

File: app/src/components/integrations/elements/integrationSettings/connectionSection/connectionSection.jsx [hunk 3]
```diff
@@ -197,7 +197,11 @@ export class ConnectionSection extends Component {
         >
           <div className={cx('integration-info-block')}>
             <div className={cx('sub-header-block')}>
-              <div className={cx('general-info')}>
+              <div
+                className={cx('general-info', {
+                  blocked: blocked && availableProjectIntegrations.length > 0,
```
Comment (Bam6ycha) [inline app/src/components/integrations/elements/integrationSettings/connectionSection/connectionSection.jsx:202]: I think that in these case we can avoid of using availableProjectIntegrations.length > 0. 
But it is not mandatory to fix.

File: app/src/components/integrations/elements/integrationSettings/connectionSection/connectionSection.jsx [hunk 4]
```diff
@@ -206,6 +210,7 @@ export class ConnectionSection extends Component {
               <div
                 className={cx('connection-block', {
                   'connection-block-failed': !connected,
+                  'connection-block-disabled': blocked && availableProjectIntegrations.length > 0,
                 })}
```
Comment (Bam6ycha) [inline app/src/components/integrations/elements/integrationSettings/connectionSection/connectionSection.jsx:214]: The same thing.
It is important to use length>0 only when we use conditional rendering.
It is not mandatory to fix.

Comment (RuslanLyubimov) [reply app/src/components/integrations/elements/integrationSettings/connectionSection/connectionSection.jsx:214]: without this, styling will be improper

File: app/src/components/integrations/elements/integrationSettings/connectionSection/connectionSection.jsx (hunk 3 above)
Comment (RuslanLyubimov) [reply app/src/components/integrations/elements/integrationSettings/connectionSection/connectionSection.jsx:202]: without this, styling will be improper

Comment (Bam6ycha) [reply app/src/components/integrations/elements/integrationSettings/connectionSection/connectionSection.jsx:202]: Okay, may be you are right.

**Review by chivekrodis — APPROVED** [review]

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**